tempesta_fw/http.c

```diff
@@ -1818,14 +1818,21 @@ tfw_http_msg_create_sibling(TfwHttpMsg *hm, struct sk_buff **skb,
 
 	/* The sibling message belongs to the same connection. */
 	shm = (TfwHttpMsg *)tfw_http_conn_msg_alloc(hm->conn);
-	if (unlikely(!shm))
+	if (unlikely(!shm)) {
+		/*
+		 * Split skb to proceed with current message @hm, or rest of
+		 * data in this connection will be attached to the message.
+		 */
+		if (type == Conn_Srv)
+			*skb = ss_skb_split(*skb, split_offset);
```

Comment:
Yeah, makes sense. But what about 2 pipelined requests? It seems we send then as one request to the server and the server returns us 2 responses - we can't find a request for the 2nd response and get #941...

Reply:
If we can't allocate a sibling message, it means that we have to drop the connection. For server connection we can use message already parsed from the connection. It's normal response and we can forward it to the client. Just need to set correct framing. And the patch is about it.

Situation is little different with client connections. As I said the connection is closed, so we drop the already processed request without further forwarding to backend server. There is no sense in splitting skb for client connections.
https://github.com/tempesta-tech/tempesta/blob/4935217018d37315f24166e38d936a0fbeb3d6bc/tempesta_fw/http.c#L2718-L2732

That's not a reason for #941

Reply:
Ok, has sense. Please add a short comment that we do not care about requests there and just drop client connections

Reply:
Ok, has sense. Please add a short comment that we do not care about requests there and just drop client connections